gw_eccentricity/eccDefinition.py

```diff
@@ -1055,26 +1071,41 @@ def interp_extrema(self, extrema_type="pericenters"):
             return self.get_interp(self.t[extrema],
                                    self.omega22[extrema])
         else:
-            raise Exception(
-                f"Sufficient number of {extrema_type} are not found."
-                " Can not create an interpolant.")
+            raise InsufficientExtrema(extrema_type, len(extrema))
 
     def check_num_extrema(self, extrema, extrema_type="extrema"):
         """Check number of extrema."""
         num_extrema = len(extrema)
         if num_extrema < 2:
-            recommended_methods = ["ResidualAmplitude", "AmplitudeFits"]
-            if self.method not in recommended_methods:
-                method_message = ("It's possible that the eccentricity is too "
-                                  f"low for the {self.method} method to detect"
-                                  f" the {extrema_type}. Try one of "
-                                  f"{recommended_methods}.")
+            # check if the waveform is sufficiently long
+            if self.approximate_num_orbits > 5:
+                # The waveform is long but the method fails to find the extrema
+                # This may happen because the eccentricity too small for the
+                # method to detect it.
+                self.probably_quasicircular = True
+            if self.probably_quasicircular and self.set_failures_to_zero:
+                debug_message(
+                    "The waveform has approximately "
+                    f"{self.approximate_num_orbits:.2f}"
+                    f" orbits but number of {extrema_type} found is "
+                    f"{num_extrema}. Since `set_failures_to_zero` is set to "
+                    f"{self.set_failures_to_zero}, no exception is raised. "
+                    "Instead the eccentricity and mean anomaly will be set to "
+                    "zero.",
+                    important=True,
+                    debug_level=0)
             else:
-                method_message = ""
-            raise Exception(
-                f"Number of {extrema_type} found = {num_extrema}.\n"
-                f"Can not build frequency interpolant through the {extrema_type}.\n"
-                f"{method_message}")
+                recommended_methods = ["ResidualAmplitude", "AmplitudeFits"]
```

Comment:
Why not ResidualFrequency as well?

Reply:
Because we recommend only Amplitude-based methods.